presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/PredicatePushdownController.java

```diff
@@ -14,15 +14,28 @@
 package io.prestosql.plugin.jdbc;
 
 import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.predicate.DiscreteValues;
 import io.prestosql.spi.predicate.Domain;
+import io.prestosql.spi.predicate.Ranges;
 
+import static io.prestosql.plugin.jdbc.JdbcMetadataSessionProperties.getDomainCompactionThreshold;
 import static java.util.Objects.requireNonNull;
 
 public interface PredicatePushdownController
 {
-    PredicatePushdownController FULL_PUSHDOWN = (session, domain) -> new DomainPushdownResult(domain, Domain.all(domain.getType()));
-    PredicatePushdownController PUSHDOWN_AND_KEEP = (session, domain) -> new DomainPushdownResult(domain, domain);
-    PredicatePushdownController DISABLE_PUSHDOWN = (session, domain) -> new DomainPushdownResult(Domain.all(domain.getType()), domain);
+    PredicatePushdownController FULL_PUSHDOWN = (session, domain) -> {
+        if (getDomainSize(domain) > getDomainCompactionThreshold(session)) {
```

Comment:
Add to commit message that you actually enable simplification of pushed down domain in this comit